src/result_processor.h

```diff
@@ -190,8 +190,10 @@ typedef struct ResultProcessor {
   void (*Free)(struct ResultProcessor *self);
 } ResultProcessor;
 
+// Get the index search context from the result processor
+#define RP_SCTX(rpctx) ((rpctx)->parent->sctx)
 // Get the index spec from the result processor
-#define RP_SPEC(rpctx) ((rpctx)->parent->sctx->spec)
+#define RP_SPEC(rpctx) (RP_SCTX(rpctx)->spec)
```

Comment:
I see that the highlighter uses this macro, so it uses the spec after we unlock it. isn't that a problem?

Reply:
Looks like it is used only to search and check if specific fields exist in the spec and that their type is INDEXFLD_T_FULLTEXT
As long as the GC can't remove/modify/move fields from the spec, looks OK

Reply:
In second thought, the highlighter doesn't hold the GIL too so no protection at this point on the spec's fields.